src/NonCumulative/Soundness/Properties/LogRel.agda

```diff
@@ -86,4 +88,620 @@ Glu-wellfounded-≡-GluΛ {j} {k} = implicit-extensionality fext (fext (λ l<k 
+  { t∶T = ctxeq-tm Γ≈Δ t∶T
+  ; ⊢T = ctxeq-tm Γ≈Δ ⊢T
+  ; krip = λ ⊢σ → krip (⊢w-resp-⊢≈ʳ ⊢σ (⊢≈-sym Γ≈Δ))
+  }
+  where open GluNe glu
+®El-resp-⊢≈ N′ (t®N , T≈N) Γ≈Δ = (®Nat-resp-⊢≈ t®N Γ≈Δ) , ctxeq-≈ Γ≈Δ T≈N
+®El-resp-⊢≈ (U {j} refl refl) t® Γ≈Δ
+  rewrite Glu-wf-gen {j} j U≤′
+        | 𝕌-wf-gen j (λ l<j → <-trans l<j (s≤s (≤-reflexive refl))) = record
+  { t∶T = ctxeq-tm Γ≈Δ t∶T
+  ; T≈  = ctxeq-≈ Γ≈Δ T≈
+  ; A∈𝕌 = A∈𝕌
+  ; rel = ®-resp-⊢≈ A∈𝕌 rel Γ≈Δ
+  }
+  where open GluU t®
+®El-resp-⊢≈ (Π eq jA x x₁ x₂) t® Γ≈Δ =
+  let Δ⊢IT = ctxeq-tm Γ≈Δ ⊢IT in record
+  { t∶T  = ctxeq-tm Γ≈Δ t∶T
+  ; a∈El = a∈El
+  ; IT   = IT
+  ; OT   = OT
+  ; ⊢IT  = Δ⊢IT
+  ; ⊢OT  = ctxeq-tm (∷-cong Γ≈Δ ⊢IT Δ⊢IT (≈-refl ⊢IT) (≈-refl Δ⊢IT)) ⊢OT
+  ; T≈   = ctxeq-≈ Γ≈Δ T≈
+  ; krip = λ ⊢σ → krip (⊢w-resp-⊢≈ʳ ⊢σ (⊢≈-sym Γ≈Δ))
+  }
+  where open GluΛ t®
+®El-resp-⊢≈ (L refl kA refl refl) t® Γ≈Δ = record
+  { t∶T  = ctxeq-tm Γ≈Δ t∶T
+  ; UT   = UT
+  ; ⊢UT  = ctxeq-tm Γ≈Δ ⊢UT
+  ; a∈El = a∈El
+  ; T≈   = ctxeq-≈ Γ≈Δ T≈
+  ; krip = λ ⊢σ → krip (⊢w-resp-⊢≈ʳ ⊢σ (⊢≈-sym Γ≈Δ))
+  }
+  where open Glul t®
+
+mutual
+  ®-swap : ∀ {i} (A≈B : A ≈ B ∈ 𝕌 i)
+           (B≈A : B ≈ A ∈ 𝕌 i) →
+           Γ ⊢ T ®[ i ] A≈B →
+           -----------------------
+           Γ ⊢ T ®[ i ] B≈A
+  ®-swap {Γ = Γ} {T} {i = i} (ne′ c≈c′) (ne c′≈c _ _) (⊢T , krip) = ⊢T , helper
+    where
+      helper : Δ ⊢w σ ∶ Γ →
+               -----------------------------------
+               Δ ⊢ T [ σ ] ≈ Ne⇒Exp (proj₁ (c′≈c (len Δ))) ∶[ 1 + i ] Se i
+      helper {Δ} {σ} ⊢σ
+        with c≈c′ (len Δ) | c′≈c (len Δ) | krip ⊢σ
+      ...  | _ , ↘u , _ | _ , _ , ↘u₁ | Tσ≈
+           rewrite Re-det ↘u ↘u₁ = Tσ≈
+  ®-swap N′ N′ T® = T®
+  ®-swap (U refl refl) (U i≡1+j j≡j′) T®
+    rewrite ≡-irrelevant i≡1+j refl = T®
+  ®-swap {_} {_} {Γ} (Π′ {j} {k} jA RT) (Π i≡maxjk jA′ RT′ j≡j′ k≡k′) record { IT = IT ; OT = OT ; ⊢IT = ⊢IT ; ⊢OT = ⊢OT ; T≈ = T≈ ; krip = krip }
+    rewrite ≡-irrelevant i≡maxjk refl
+          | 𝕌-wf-gen j (ΠI≤′ j k refl)
+          | 𝕌-wf-gen k (ΠO≤′ j k refl)
+          | Glu-wf-gen j (ΠI≤′ j k refl)
+          | Glu-wf-gen k (ΠO≤′ j k refl) = record
+    { IT   = IT
+    ; OT   = OT
+    ; ⊢IT  = ⊢IT
+    ; ⊢OT  = ⊢OT
+    ; T≈   = T≈
+    ; krip = λ ⊢σ → let open ΠRel (krip ⊢σ) in record
+      { IT-rel = ®-swap jA jA′ IT-rel
+      ; OT-rel = λ s® a∈ → OT-helper a∈ s® OT-rel
+      }
+    }
+    where OT-helper : (a∈′ : a ∈′ El j jA′) →
+                      Δ ⊢ s ∶ IT [ σ ] ®[ j ] a ∈El jA′ →
+                      (∀ {s a} → Δ ⊢ s ∶ IT [ σ ] ®[ j ] a ∈El jA →
+                        (a∈ : a ∈′ El j jA) →
+                        Δ ⊢ OT [ σ , s ∶ IT ↙ j ] ®[ k ] ΠRT.T≈T′ (RT a∈)) →
+                      --------------------------------------------------------------
+                      Δ ⊢ OT [ σ , s ∶ IT ↙ j ] ®[ k ] ΠRT.T≈T′ (RT′ a∈′)
+          OT-helper a∈′ s® OT-rel
+            with El-sym jA′ jA a∈′
+          ... | a∈
```

Comment:
style: customarily, the pipe on the next line should  align with the last character of the previous line. Emacs can do it quite easily.